tests/components/wallbox/test_init.py

```diff
@@ -31,135 +27,9 @@
 )
 
 
-async def test_wallbox_setup_entry(hass: HomeAssistant):
-    """Test Wallbox Setup."""
-    with requests_mock.Mocker() as m:
-        m.get(
-            "https://api.wall-box.com/auth/token/user",
-            text='{"jwt":"fakekeyhere","user_id":12345,"ttl":145656758,"error":false,"status":200}',
-            status_code=200,
-        )
-        m.get(
-            "https://api.wall-box.com/chargers/status/12345",
-            text='{"Temperature": 100, "Location": "Toronto", "Datetime": "2020-07-23", "Units": "Celsius"}',
-            status_code=200,
-        )
-        assert await wallbox.async_setup_entry(hass, entry)
-
-    with requests_mock.Mocker() as m, raises(ConnectionError):
-        m.get(
-            "https://api.wall-box.com/auth/token/user",
-            text='{"jwt":"fakekeyhere","user_id":12345,"ttl":145656758,"error":false,"status":404}',
-            status_code=404,
-        )
-        assert await wallbox.async_setup_entry(hass, entry) is False
-
-
 async def test_wallbox_unload_entry(hass: HomeAssistant):
     """Test Wallbox Unload."""
-    hass.data[DOMAIN] = {"connections": {entry.entry_id: entry}}
-
-    assert await wallbox.async_unload_entry(hass, entry)
-
-    hass.data[DOMAIN] = {"fail_entry": entry}
-
-    with pytest.raises(KeyError):
-        await wallbox.async_unload_entry(hass, entry)
-
-
-async def test_get_data(hass: HomeAssistant):
-    """Test hub class, get_data."""
-
-    station = ("12345",)
-    username = ("test-username",)
-    password = "test-password"
-
-    hub = wallbox.WallboxHub(station, username, password, hass)
-
-    with requests_mock.Mocker() as m:
-        m.get(
-            "https://api.wall-box.com/auth/token/user",
-            text='{"jwt":"fakekeyhere","user_id":12345,"ttl":145656758,"error":false,"status":200}',
-            status_code=200,
-        )
-        m.get(
-            "https://api.wall-box.com/chargers/status/('12345',)",
-            json=test_response,
-            status_code=200,
-        )
-        assert await hub.async_get_data()
-
-
-async def test_get_data_rounding_error(hass: HomeAssistant):
-    """Test hub class, get_data with rounding error."""
-
-    station = ("12345",)
-    username = ("test-username",)
-    password = "test-password"
-
-    hub = wallbox.WallboxHub(station, username, password, hass)
-
-    with requests_mock.Mocker() as m:
-        m.get(
-            "https://api.wall-box.com/auth/token/user",
-            text='{"jwt":"fakekeyhere","user_id":12345,"ttl":145656758,"error":false,"status":200}',
-            status_code=200,
-        )
-        m.get(
-            "https://api.wall-box.com/chargers/status/('12345',)",
-            json=test_response_rounding_error,
-            status_code=200,
-        )
-        assert await hub.async_get_data()
-
-
-async def test_authentication_exception(hass: HomeAssistant):
```

Comment:
What happens when there's an authentication error during config entry setup? That doesn't seem handled.

Reply:
That test is in the test_config_flow file (test_form_cannot_authenticate), I test how a 403 error on the authentication is handled.

Reply:
Right, but it can happen during config entry set up too. So ideally we should have both code and tests that handles and tests this.